### ISSUE TITLE
Do I need to set larger restart period?

### ISSUE THREAD

**Issue by twmht** [loshchil/AdamW-and-SGDW#5]
Hi,

in the original sgdr paper (https://arxiv.org/abs/1608.03983), you suggest to restart after 10 epochs (Te=10) in the first restart.

but in this paper, you give the example to restart after 100 epochs.

is it very sensitive to the restart period after  `fixing weight decay` ?

Since I have trained both SGD and SGDW when TE=10, but only SGD can achieve lowest validation error when first restart.  So I am not sure if I have to set lager restart when using SGDW. 

Any idea?

**Comment by loshchil** [comment]
As you can see from the figure you posted in a previous ticket, the optimal hyperparameter settings for SGD and SGDW are (likely to be) different. Thus, you may need to first figure out optimal weight decay factor (or preferably, normalized weight decay - see the paper) and learning rate for SGDW. SGDW and SGD are not expected to be very different in performance when tested at optimal hyperparameter settings.
 
The main purpose of restarts is to improve anytime performance, i.e., be able to get better solutions faster. The budget given for the first restart might be something between say 1% and 20% of the total budget of epochs.

**Comment by twmht** [comment]
thank you!
